std_execution.bs

```diff
@@ -5161,197 +5161,86 @@ are all well-formed.
-                `op_state3`. `op_state3` is saved as a part of `op_state2`. It then calls `execution::start(op_state3)`. If any of these throws an exception, it catches it and calls `execution::set_error(std::move(out_r'), current_exception())`. If any of these expressions would be ill-formed, `execution::set_stopped(r)` is ill-formed.
-
-        2. `execution::let_stopped(s, f)` returns a sender `s2` such that:
-        
-            1. If the expression `execution::connect(s, r)` is ill-formed, `execution::connect(s2, out_r)` is ill-formed.
-            
-            2. Otherwise, let `op_state2` be the result of `execution::connect(s, r)`. `execution::connect(s2, out_r)` returns an operation state `op_state` that stores `op_state2`. `execution::start(op_state)` is expression-equivalent to `execution::start(op_state2)`.
+                1. We denote by the expression <code><i>completion-sigs</i>&lt;Snd, Env></code> the set set of signature types in the <code><i>type-list</i></code> equivalent to the template arguments of the `completion_signatures` class returned by `execution::get_completion_signatures(declval<Snd>(), declval<Env>)`.
 
-        3. Given some expression `e`, let `E` be `decltype((e))`, and let `S3` be `invoke_result_t<F>`. If either `sender<S, E>` or `sender<S3, E>` is `false`, the type of `tag_invoke(get_completion_signatures, s2, e)` shall be equivalent to `dependent_completion_signatures<E>`. Otherwise, let `As...` be the set of types in the <code><i>type-list</i></code> named by <code>value_types_of_t&lt;S, E, <i>set-value-signature</i>, <i>type-list</i>></code>, and let `Ws...` be the types in the <code><i>type-list</i></code> named by <code>error_types_of_t&lt;S, E, <i>type-list</i>></code>, where <code><i>set-value-signature</i></code> is the alias template:
-        
-            <pre highlight="c++">
-            template &lt;class... Ts>
-              using <i>set-value-signature</i> = set_value_t(Ts...);
-            </pre>
-
-            1. Let `Vs...` be the list of types in the <code><i>type-list</i></code> named by <code>value_types_of_t&lt;S3, E, <i>set-value-signature</i>, <i>type-list</i>></code>, and let `Es...` be the list of types in the <code><i>type-list</i></code> named by <code>error_types_of_t&lt;S3, E, <i>type-list</i>></code>.
+                2. Let `base_sigs...` be the subset of types from <code><i>completion-sigs&lt;S, no_env></code> that do not contain any signature of the form <code><i>set-cpo</i>(...)</code>.
 
-            4. Let `Us...` be the set of unique types in `[As..., Vs...]` and let `Ws...` be the unique set of types in `[exception_ptr, Es...]`,.
+                3. Let `sigs...` be the unique set of types <code>[base_sigs..., <i>completion-sigs</i>&lt;S3, e>..., set_error_t(exception_ptr)]</code>.
```

Comment:
With the `S3` pack expansion, you'll have to do something like:

> for each <code>S3<i><sub>i</sub></i></code> in `S3s...`, blah blah <code>base_sigs..., <i>completion-sigs</i>&lt;S3<i><sub>0</sub></i>, e>, <i>completion-sigs</i>&lt;S3<i><sub>1</sub></i>, e>... <i>completion-sigs</i>&lt;S3<i><sub>n-1</sub></i>, e>, set_error_t(exception_ptr)</code>, where <code><i>n</i></code> is `sizeof...(S3s)`.

Also, by `e` did you mean `E`?